src/OpenLoco/Windows/ScenarioSelect.cpp

```diff
@@ -0,0 +1,503 @@
+            for (int i = 0; i < 5; i++)
+            {
+                if (self->widgets[widx::tab0 + i].type == widget_type::none)
+                {
+                    selectedTab = i;
+                    break;
+                }
+            }
+        }
+
+        self->current_tab = selectedTab;
+
+        initList(self);
+
+        return self;
+    }
+
+    // 0x00443995
+    static void prepareDraw(window* self)
+    {
+        self->activated_widgets &= ~((1 << widx::tab0) | (1 << widx::tab1) | (1 << widx::tab2) | (1 << widx::tab3) | (1 << widx::tab4));
+        self->activated_widgets |= (1ULL << (self->current_tab + static_cast<uint8_t>(widx::tab0)));
+    }
+
+    // 0x004439AF
+    static void draw(window* self, Gfx::drawpixelinfo_t* dpi)
+    {
+        Gfx::drawRectInset(dpi, self->x, self->y + 20, self->width, 41, self->colours[0], 0);
+
+        // Draw widgets.
+        self->draw(dpi);
+
+        static const string_id scenarioGroupIds[] = {
+            StringIds::scenario_group_beginner,
+            StringIds::scenario_group_easy,
+            StringIds::scenario_group_medium,
+            StringIds::scenario_group_challenging,
+            StringIds::scenario_group_expert,
+        };
+
+        // Draw tab captions.
+        for (int i = 0; i < 5; i++)
+        {
+            widget_t& widget = self->widgets[widx::tab0 + i];
+            if (widget.type == widget_type::none)
+                continue;
+
+            const auto offset = self->current_tab == i ? 1 : 0;
+            auto origin = Gfx::point_t(widget.mid_x() + self->x, widget.mid_y() + self->y - 3 - offset);
+            const string_id caption = scenarioGroupIds[i];
+
+            Gfx::drawStringCentredWrapped(dpi, &origin, 87, Colour::black, StringIds::wcolour2_stringid, &caption);
+        }
+
+        // Scenario selected?
+        if (self->info == 0 || self->info == 0xFFFFFFFF)
+            return;
+
+        using namespace scenariomgr;
+        auto scenarioInfo = reinterpret_cast<ScenarioIndexEntry*>(self->info);
+
+        // Check if current currency object needs to be changed.
+        registers regs;
+        regs.ebp = reinterpret_cast<int32_t>(&scenarioInfo->currency);
+        bool changeCurrency = (call(0x004720EB, regs) & (X86_FLAG_CARRY << 8)) == 0;
+
+        // Load currency object if needed.
+        if (changeCurrency)
+        {
+            // Unload current object
+            registers regs2;
+            regs2.ebp = 0x0011264A4; // currencyMeta
+            call(0x00471FF8, regs2); // unload object
+
+            // Load required object
+            registers regs3;
+            regs3.ebp = (uintptr_t)scenarioInfo->currency.var_04;
+            call(0x00471BCE, regs3);
+            call(0x0047237D); // reset_loaded_objects
+            call(0x0046E07B); // load currency gfx
+        }
+
+        // Scenario name
+        {
+            const int16_t x = self->x + self->widgets[widx::list].right + 89;
```

Comment:
If we define the top/left coordinate first and then increment the Y position per element drawn, we can keep the height information better documented.

```
x = left+4
y = top+8
width = 60;

{
    drawStringCentred(str, x+width/2, y)
    y += 15;
}

{
   drawImage(x, y)
   y+=128;
}

{
    stringHeight= drawOtherString(x, y);
    y+= stringHeight;
}
```

Reply:
This is something I prefer to clean up at some other point. I will keep that in mind for another PR, though.

Reply:
Add a TODO for it then 👍